modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -1342,6 +1342,14 @@ private ObjectField _updateObjectField(
 					"object definition and cannot be changed");
 		}
 
+		if (Objects.equals(required, true) &&
+			!Objects.equals(oldObjectField.getReadOnly(), readOnly)) {
+
+			throw new ObjectFieldReadOnlyException(
+				"The read only setting of an required object field cannot be " +
+					"changed");
+		}
+
```

Comment:
@larissacribeiro Based on the ticket description, the error should be thrown because:

> Required fields cannot be read-only. So disable the read-only if the field is required.

So I think we should **also** add the validation when creating a new Object Field, not just when updating, since the user can create a Object Field via Headless API.

That said, I believe we could add the new logic to the existing method `validateRequired`:

```
if (required && !Objects.equals(readOnly, ObjectFieldConstants.READ_ONLY_FALSE)) {
	throw new ObjectFieldRequiredException("Required object field cannot be read only");
}
```
In this case I am checking if the field is `required`. If so, the `readOnly` property must always be `false`.

Reply:
Thanks, @carolmariaabb! I've been trying to make this work on the `validateRequired` but I couldn't, instead I made it on the `validateReadOnlyAndReadOnlyConditionExpression` and in this one it works, which I believe will give the same results of what you suggested. Please, tell me if you think otherwise! 😃